app/src/main/java/com/google/samples/apps/sunflower/viewmodels/PlantDetailViewModel.kt

```diff
@@ -17,22 +17,42 @@
 package com.google.samples.apps.sunflower.viewmodels
 
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
 import androidx.lifecycle.viewModelScope
 import com.google.samples.apps.sunflower.BuildConfig
 import com.google.samples.apps.sunflower.PlantDetailFragment
 import com.google.samples.apps.sunflower.data.GardenPlantingRepository
 import com.google.samples.apps.sunflower.data.PlantRepository
+import com.squareup.inject.assisted.AssistedInject
+import com.squareup.inject.assisted.Assisted
 import kotlinx.coroutines.launch
 
 /**
  * The ViewModel used in [PlantDetailFragment].
  */
-class PlantDetailViewModel(
+class PlantDetailViewModel @AssistedInject constructor(
     plantRepository: PlantRepository,
     private val gardenPlantingRepository: GardenPlantingRepository,
-    private val plantId: String
+    @Assisted private val plantId: String
 ) : ViewModel() {
 
+    @AssistedInject.Factory
```

Comment:
Move this added code to the bottom of the file as it's not that important and shouldn't be what you focus on when you open this file